sdk/src/azure/platform/az_curl.c

```diff
@@ -465,7 +468,9 @@ _az_http_client_curl_setup_url(CURL* ref_curl, az_http_request const* request)
   az_span request_url = { 0 };
   // get request_url. It will have the size of what it has written in it only
   AZ_RETURN_IF_FAILED(az_http_request_get_url(request, &request_url));
-  int32_t request_url_size = _az_span_url_encode_calc_length(request_url);
+  // TODO: url-encode, encode only query parameters.
```

Comment:
I would prefer if we fixed this now rather than leaving it as a TODO. 

And, if we really want to revert it, do a clean revert of the previous commit (using `git revert`) if possible.

Reply:
The fix might be a huge refactor.
We don´t want a revert because we do want then changes in the credential.

We need to merge this to avoid shipping a broken transport adapter.